File: Python/backend_flask_archi_feature/author/db/author_inmemory.py
from datetime import date
from typing import List, Optional

class AuthorInMemory:
    def __init__(self, id: int, first_name: str, last_name: str, birth_date: date,
                 nationality: Optional[str] = None, death_date: Optional[date] = None,
                 biography: Optional[str] = None, books: Optional[List[int]] = None):
        self.id = id
        self.first_name = first_name
        self.last_name = last_name
        self.birth_date = birth_date
        self.nationality = nationality
        self.death_date = death_date
        self.biography = biography
        self.books = books or []

File: Javascript/backend_nest/src/infra/db/inmemory/entities/AuthorInMemory.ts
export class AuthorInMemory {
  id: number;
  firstName: string;
  lastName: string;
  birthDate: string;
  nationality: string;
  deathDate: string | null;
  biography?: string;

  constructor(
    id: number,
    firstName: string,
    lastName: string,
    birthDate: string,
    nationality: string,
    deathDate: string | null,
    biography?: string
  ) {
    this.id = id;
    this.firstName = firstName;
    this.lastName = lastName;
    this.birthDate = birthDate;
    this.nationality = nationality;
    this.deathDate = deathDate;
    this.biography = biography;
  }
}
